engine/physics/src/box2d/Box2D/Collision/Shapes/b2GridShape.cpp

```diff
@@ -149,11 +149,25 @@ uint32 b2GridShape::GetCellVertices(uint32 index, b2Vec2* vertices) const
     float32 xScale = flags.m_FlipHorizontal ? -1.0f : 1.0f;
     float32 yScale = flags.m_FlipVertical ? -1.0f : 1.0f;
 
+    float32 tmpX = 0.0;
+
     for (uint32 i = 0; i < hull.m_Count; ++i)
     {
         vertices[i] = m_hullSet->m_vertices[hull.m_Index + i];
-        vertices[i].x *= xScale * m_cellWidth;
-        vertices[i].y *= yScale * m_cellHeight;
+        if (flags.m_Rotate90)
+        {
+            // Clockwise rotation (x, y) -> (y, -x)
+            // Also tile isn't necessary a square, that's why we should swap width and height 
```

Comment:
What would it mean to have a non-square tile and rotate it?
I don't think we've discussed non-square tiles and how they should work at all.
So it's a little bit strange to add support for it (the elaborate comment makes it feel like we expect it to work in the future)

Reply:
I didn't have a choice ) to avoid crashed I need to add support of such rotation, so, for now tile just stretch the cell (for both physics and graphics)